internal/provider/provider.go

```diff
@@ -93,8 +101,99 @@ func (pm *NuoDbaasProviderModel) GetSkipVerify() bool {
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true, //nolint:gosec // Reduced security at the demand of the user.
+			},
+		}
+	}
+	return client
+}
+
+func (pm *NuoDbaasProviderModel) getAuthHeader() string {
+	if pm.GetToken() != "" {
+		return "Bearer " + pm.GetToken()
+	} else if pm.GetUser() != "" {
+		auth := pm.GetUser() + ":" + pm.GetPassword()
+		return "Basic " + base64.StdEncoding.EncodeToString([]byte(auth))
+	}
+	return ""
+}
+
+func (pm *NuoDbaasProviderModel) prepareRequest(req *http.Request) {
+	if req.Header.Get("Authorization") == "" {
+		if authHeader := pm.getAuthHeader(); authHeader != "" {
+			req.Header.Set("Authorization", authHeader)
+		}
+	}
+}
+
+func (pm *NuoDbaasProviderModel) CreateClient() (openapi.ClientInterface, error) {
+	return openapi.NewClient(pm.GetUrlBase(),
+		openapi.WithHTTPClient(pm.getHttpClient()),
+		openapi.WithRequestEditorFn(func(ctx context.Context, req *http.Request) error {
+			pm.prepareRequest(req)
+			return nil
+		}))
+}
+
+func (pm *NuoDbaasProviderModel) buildSseRequest(ctx context.Context, path string) (*http.Request, error) {
+	url := strings.TrimSuffix(pm.GetUrlBase(), "/") + "/" + strings.TrimPrefix(path, "/")
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, http.NoBody)
+	if err != nil {
+		return nil, err
+	}
+	pm.prepareRequest(req)
+	return req, nil
+}
+
+func (pm *NuoDbaasProviderModel) createSseClient(ctx context.Context) *sse.Client {
+	// Copy default SSE client and replace HTTP client
+	sseClient := *sse.DefaultClient
+	sseClient.HTTPClient = pm.getHttpClient()
+	// Register callback on context cancellation to close response reader.
+	// This unblocks any concurrent read to allow the goroutine dispatching
+	// SSE messages to terminate.
+	var readerRef atomic.Value
+	context.AfterFunc(ctx, func() {
+		if reader, ok := readerRef.Load().(io.Closer); ok {
+			tflog.Debug(ctx, "Closing SSE message reader")
+			_ = reader.Close()
+		}
+	})
+	// Use response validator to set or update reader
+	sseClient.ResponseValidator = func(resp *http.Response) error {
+		err := sse.DefaultValidator(resp)
+		if err == nil {
+			readerRef.Store(resp.Body)
+			// If context is already done, close reader
+			if ctx.Err() != nil {
+				resp.Body.Close()
+			}
+		}
+		return err
+	}
+	// Set max backoff interval for reconnects to polling interval
+	sseClient.Backoff.MaxInterval = framework.POLLING_INTERVAL
+	return &sseClient
+}
+
+func (pm *NuoDbaasProviderModel) ConsumeEvents(ctx context.Context, path string, callback func(sse.Event)) error {
+	// Build SSE request
+	req, err := pm.buildSseRequest(ctx, path)
+	if err != nil {
+		return err
+	}
+	// Create SSE client and connection
+	sseClient := pm.createSseClient(ctx)
+	sseConnection := sseClient.NewConnection(req)
+	// Register callback and consume SSE messages synchronously until a
+	// non-retriable error occurs
+	sseConnection.SubscribeToAll(callback)
+	return sseConnection.Connect()
```

Comment:
This will always return an error because `Connect()` blocks until the context is done. The callers should inspect the context to determine if the error is on the initial connect.

Should we follow the client guidelines instead?

```golang
if err := conn.Connect(); !errors.Is(err, context.Canceled) {
    return err
}
// suppress the error as the context is canceled
return nil
```

Reply:
I'll look into this.